src/chatproviderservice/src/Client/ChatProviderServiceClient.lua

```diff
@@ -69,16 +77,30 @@ end
 
 --[=[
 	Sends a system message to the provided TextChannel.
-	@param channel TextChannel
-	@param message string
-	@param color Color3?
+	@param encodedMessageData string
+	@param channel TextChannel?
 ]=]
-function ChatProviderServiceClient:SendSystemMessage(channel: TextChannel, message: string, color: Color3?)
-	assert(typeof(channel) == "Instance" and channel.ClassName == "TextChannel", "[ChatProviderServiceClient.SendSystemMessage] - Bad channel")
+function ChatProviderServiceClient:SendSystemMessage(message: string, encodedMessageData: string?, channel: TextChannel?)
 	assert(typeof(message) == "string", "[ChatProviderServiceClient.SendSystemMessage] - Bad message")
-	assert(typeof(color) == "Color3" or color == nil, "[ChatProviderServiceClient.SendSystemMessage] - Bad color")
 
-	channel:DisplaySystemMessage(message, color and color:ToHex())
+	if not channel then
+		local channels = TextChatService:FindFirstChild("TextChannels")
```

Comment:
I usually put in utility function for readability, but up to you.

Reply:
just :_getDefaultTextChannel() or TextChannelUtils.getDefaultTextChannel() but ehhh

Reply:
resolved in [49fdd74](https://github.com/Quenty/NevermoreEngine/pull/520/commits/49fdd7495ee83b23aa8e042f7055d080e50f9687)